goalc/compiler/Compiler.h

```diff
@@ -21,6 +21,7 @@
 
 #include "third-party/fmt/color.h"
 #include "third-party/fmt/core.h"
+#include "third-party/json.hpp"
```

Comment:
can we avoid having the `json` include in this file? It will make all the compiler stuff slow to compile because the json library is a huge mess that slows everything down.